Assets/Scripts/Manager/GameManager.cs

```diff
@@ -9,18 +14,78 @@ public class GameManager : MonoBehaviour {
 
 	public bool trueEnding;
 
+	public string currentLevel;
+	public bool[] secretItemFound;
+
+	public bool firstRun;
+	public bool[] textTriggered;
+
+	public GameObject saveText;
+
+	private static String[] canSaveScenes = {"OpenScene", "1a", "1b", "2a", "2b", "3a", "3b", "4a-i", "4a-ii", "4b"};
+
+	private static String[] cannotRestartScenes = {"OpenScene"};
+
 	//Awake is always called before any Start functions
 	void Awake()
 	{
 		if (instance == null) {
+			//Game just start
+			DontDestroyOnLoad (transform.root.gameObject);
 			instance = this;
+			instance.secretItemFound = new bool[5];
+			trueEnding = false;
+			initSceneFirstRun();
 		} else if (instance != this) {
-			Destroy (gameObject);    
+			Destroy (gameObject);
+			Debug.Log ("destroy Game Manager duplicate on Awake");
+
+			if (!instance.currentLevel.Equals (SceneManager.GetActiveScene ().name)) {
+				// Just enter a new scene
+				initSceneFirstRun();
+			} else {
+				initScene ();
+				instance.firstRun = false;
+			}
 		}
-			
-		DontDestroyOnLoad(gameObject);
-		trueEnding = false;
-		//InitGame();
+	}
+
+	void Start(){
+	}
+
+	void Update(){
+		if (Input.GetKeyDown(KeyCode.S)){
```

Comment:
why are we getting the key of "S" to save??? Was this for testing? Can we remove it?